modules/grafana-agent-tempo/variables.tf

```
@@ -11,6 +11,11 @@ variable "tempo_endpoint_retry_on_failure" {
   default     = false
   description = "Whether to enable 'Retry on failure' for the default endpoint."
 }
+variable "tempo_endpoint_protocol" {
+  type        = string
+  default     = "grpc"
+  description = "The protocol to use when sending to the (first) endpoint. grpc/http."
```

If you want to be even more strict:
```suggestion
  description = "The protocol to use when sending to the (first) endpoint. grpc/http."
  
  validation {
    condition     = contains(["grpc", "http"], var.tempo_endpoint_protocol)
    error_message = "The tempo_endpoint_protocol must be \"grpc\" or \"http\"."
  } 
```

---

I want to avoid requiring further patching in case grafana-agent starts supporting more push protocols.